File: 30-Days-of-Code/Day-6-Lets-Review/Fortran/solution.f
INTEGER :: num
CHARACTER(10000) :: line
CHARACTER(:),ALLOCATABLE :: firstHalf, secondHalf
CHARACTER(10001) :: output

READ *, num

iloop: do i = 1, num
    READ (*,'(a)') line
    !WRITE (*, '(a)') TRIM(line)
    
    jloop: do j = 1, LEN_TRIM(line)
        IF (MOD(j, 2)==1) THEN
            firstHalf = firstHalf // line(j:j)
            !PRINT *, firstHalf
        ELSE
            secondHalf = TRIM(secondHalf) // line(j:j)
            !PRINT *, secondHalf
        ENDIF
    end do jloop
    
    output = TRIM(firstHalf) // " " // secondHalf
    
    WRITE(*,'(a)') TRIM(output)
    
    line = ""
    firstHalf = ""
    secondHalf = ""
    
end do iloop

end

File: 30-Days-of-Code/Day-6-Lets-Review/Lua/solution.lua
n = io.read("*number", "*l")

for _ = 1, n
do
    s = io.read("*l")
    
    firstHalf = ""
    secondHalf = ""
    
    for i = 1, #s
    do
        c = s:sub(i, i)
        
        if i % 2 ~= 0 then
            firstHalf = firstHalf .. c
        else
            secondHalf = secondHalf .. c
        end
    end
    print(string.format("%s %s", firstHalf, secondHalf))
end
